Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -706,4 +706,23 @@ def test_irb
 
     assert_match "Interactive Homebrew Shell", cmd("irb", irb_test)
   end
+
+  def test_analytics
+    assert_match "Invalid usage", cmd_fail("analytics", "on", "off")
+    assert_match "Invalid usage", cmd_fail("analytics", "testball")
+    assert_match "Analytics is enabled", cmd("analytics")
+    assert_match "Analytics is disabled",
+      cmd("analytics", "HOMEBREW_NO_ANALYTICS" => "1")
+
+    HOMEBREW_REPOSITORY.cd do
+      shutup do
+        system "git", "init"
+      end
+    end
```

Comment:
Hmmm. I had thought that the Git repository needed to be initialized before running `cmd("analytics", "regenerate-uuid")`, but this `cmd` seems to run smoothly regardless. I would have expected an error and I thought I remembered receiving one last week...

It does need to be initialized before toggling between `brew analytics on` and `brew analytics off`; running those `cmd`s without first initializing the repo does result in this error:

```
error: could not lock config file /tmp/homebrew-tests-20160726-97875-1wboued/prefix/.git/config:
No such file or directory
Error: Failure while executing:
git config --file=/tmp/homebrew-tests-20160726-97875-1wboued/prefix/.git/config
--replace-all homebrew.analyticsdisabled false[true]
```

In any case, I guess there's no solid reason for the Git repository to be initialized in the middle of the test. It seems to be fine to initialize it at the beginning. 😄